src/nextjournal/clerk/cljs_libs.clj

```diff
@@ -68,36 +71,28 @@
                                             [::orphan]))))
             nil))))))
 
-(defn all-ns []
+(defn all-ns [cljs-graph]
   (remove #(= ::orphan %) (tnsd/topo-sort @cljs-graph)))
 
-(defn remove-ns [ns]
-  (swap! cljs-graph (fn [graph]
-                      (-> graph
-                          (tnsd/remove-all ns))))
-  nil)
-
-(defn update-blocks [doc]
-  (update doc :blocks (fn [blocks]
-                        (concat
-                         (let [resources (keep ns->resource (all-ns))]
-                           (map (fn [resource]
-                                  (let [code-str (slurp resource)]
-                                    {:type :code
-                                     :text (pr-str `(nextjournal.clerk/eval-cljs-str ~code-str))
-                                     :result {:nextjournal/value (v/eval-cljs-str code-str)}
-                                     :settings #:nextjournal.clerk{:visibility {:code :hide, :result :hide}}}))
-                                resources))
-                         blocks))))
+(defn prepend-required-cljs [doc]
+  (let [g (new-graph)]
+    (w/postwalk (fn [v]
+                  (when-let [cljs-ns (some-> v :nextjournal/viewer :require-cljs)]
+                    (require-cljs* g cljs-ns))
+                  v)
+                doc)
+    (into (omap/ordered-map :effects (let [resources (keep ns->resource (all-ns g))]
+                                       (mapv (fn [resource]
+                                               (let [code-str (slurp resource)]
+                                                 (v/->ViewerEval `(load-string ~code-str))))
```

Comment:
Do we need to handle the evaluator=cherry case? Before it was implicitly handled during presentation.